### PR TITLE
[DX] Remove *AnnotationIncorrectNullableRector rules as works with unreliable docblocks and can have 2 solutions

### PR DESCRIPTION
**Pull request by TomasVotruba**
These rules work with unreliable docblocks and can have 2 solutions.

Since Rector 0.15 we declare type stricness and avoid docblock changes, as unreliable and could lead to incorrect assumptions: https://getrector.com/blog/new-in-rector-015-complete-safe-and-known-type-declarations

These are one of few forgotten rules. Instead handle manually and let PHPStan help you decide :+1:

### CODE REVIEW

File: rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php [hunk 1]
```diff
@@ -1,187 +0,0 @@
...
-use Rector\TypeDeclaration\NodeAnalyzer\ParamAnalyzer;
-use Rector\TypeDeclaration\PhpDocParser\ParamPhpDocNodeFactory;
-use Rector\VersionBonding\Contract\MinPhpVersionInterface;
-use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
-use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
-
-/**
- * @see \Rector\Tests\TypeDeclaration\Rector\ClassMethod\ParamAnnotationIncorrectNullableRector\ParamAnnotationIncorrectNullableRectorTest
- */
-final class ParamAnnotationIncorrectNullableRector extends AbstractRector implements MinPhpVersionInterface
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php:29]: There are 3 rules that fix incorrect nullable:

```
ReturnAnnotationIncorrectNullableRector
ParamAnnotationIncorrectNullableRector
VarAnnotationIncorrectNullableRector
```

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php:29]: I see, I'll add them to this removal list :+1:  Thanks

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php:29]: actually, I can see these rules are actually good, this example verify based on typed existing param, eg:

```diff
     /**
-     * @param \DateTime[] $dateTimes
+     * @param \DateTime[]|null $dateTimes
      */
     public function setDateTimes(?array $dateTimes): self
```

**Comment by samsonasik** [comment]
@TomasVotruba actually, I can see these rules are actually good, this example verify based on typed existing param, eg:

```diff
     /**
-     * @param \DateTime[] $dateTimes
+     * @param \DateTime[]|null $dateTimes
      */
     public function setDateTimes(?array $dateTimes): self
```
probably you can reconsider it?

**Comment by TomasVotruba** [comment]
I was thinking about the usefulness and they can preserve the state like example. The question is, often the nullability is not desired and code should be changed differently:

```diff
     /**
     * @param \DateTime[] $dateTimes
      */
-     public function setDateTimes(?array $dateTimes): self
+     public function setDateTimes(array $dateTimes): self
```

Same applies for var/return cases.

Instead of silently preserving wrong state, it should be handled by developer to fits exact needs. That's why I think it's better let PHPStan report these and people to decide what way should be taken.

**Pull request merged**